source_sdk/user_cmd.h

```diff
@@ -12,6 +12,44 @@ struct user_cmd
     float sidemove;
     float upmove;
     int buttons;
+    unsigned char impulse;
+    int weapon_select;
+    int weapon_subtype;
+    int random_seed;
+    short mouse_dx;
+    short mouse_dy;
+};
+
+// not sure if this is the best place, but since user_cmd->buttons comes from here
+// i think it comes in handy being together
+enum in_buttons 
+{
```

Comment:
You can move this somewhere else if you think it't better

Reply:
Makes sense here